src/de/mod.rs

```diff
@@ -820,49 +937,118 @@ mod tests {
         assert_eq!(crate::from_str(r#" " " "#), Ok((" ", 5)));
         assert_eq!(crate::from_str(r#" "👏" "#), Ok(("👏", 8)));
 
-        // no unescaping is done (as documented as a known issue in lib.rs)
-        assert_eq!(crate::from_str(r#" "hel\tlo" "#), Ok(("hel\\tlo", 11)));
-        assert_eq!(crate::from_str(r#" "hello \\" "#), Ok(("hello \\\\", 12)));
+        fn s(s: &'static str) -> heapless::String<1024> {
+            core::str::FromStr::from_str(s).expect("Failed to create test string")
+        }
+
+        fn from_str_test<'de, T: serde::Deserialize<'de>>(
+            s: &'de str,
+        ) -> super::Result<(T, usize)> {
+            crate::from_str_escaped(s, &mut [0; 16])
+        }
 
         // escaped " in the string content
-        assert_eq!(crate::from_str(r#" "foo\"bar" "#), Ok((r#"foo\"bar"#, 12)));
+        assert_eq!(from_str_test(r#" "foo\"bar" "#), Ok((s(r#"foo"bar"#), 12)));
         assert_eq!(
-            crate::from_str(r#" "foo\\\"bar" "#),
-            Ok((r#"foo\\\"bar"#, 14))
+            from_str_test(r#" "foo\\\"bar" "#),
+            Ok((s(r#"foo\"bar"#), 14))
         );
         assert_eq!(
-            crate::from_str(r#" "foo\"\"bar" "#),
-            Ok((r#"foo\"\"bar"#, 14))
+            from_str_test(r#" "foo\"\"bar" "#),
+            Ok((s(r#"foo""bar"#), 14))
         );
-        assert_eq!(crate::from_str(r#" "\"bar" "#), Ok((r#"\"bar"#, 9)));
-        assert_eq!(crate::from_str(r#" "foo\"" "#), Ok((r#"foo\""#, 9)));
-        assert_eq!(crate::from_str(r#" "\"" "#), Ok((r#"\""#, 6)));
+        assert_eq!(from_str_test(r#" "\"bar" "#), Ok((s(r#""bar"#), 9)));
+        assert_eq!(from_str_test(r#" "foo\"" "#), Ok((s(r#"foo""#), 9)));
+        assert_eq!(from_str_test(r#" "\"" "#), Ok((s(r#"""#), 6)));
 
         // non-excaped " preceded by backslashes
         assert_eq!(
-            crate::from_str(r#" "foo bar\\" "#),
-            Ok((r#"foo bar\\"#, 13))
+            from_str_test(r#" "foo bar\\" "#),
+            Ok((s(r#"foo bar\"#), 13))
+        );
+        assert_eq!(
+            from_str_test(r#" "foo bar\\\\" "#),
+            Ok((s(r#"foo bar\\"#), 15))
         );
         assert_eq!(
-            crate::from_str(r#" "foo bar\\\\" "#),
-            Ok((r#"foo bar\\\\"#, 15))
+            from_str_test(r#" "foo bar\\\\\\" "#),
+            Ok((s(r#"foo bar\\\"#), 17))
         );
         assert_eq!(
-            crate::from_str(r#" "foo bar\\\\\\" "#),
-            Ok((r#"foo bar\\\\\\"#, 17))
+            from_str_test(r#" "foo bar\\\\\\\\" "#),
+            Ok((s(r#"foo bar\\\\"#), 19))
+        );
+        assert_eq!(from_str_test(r#" "\\" "#), Ok((s(r#"\"#), 6)));
+    }
+
+    #[test]
+    fn tuple_of_str() {
+        fn s(s: &'static str) -> heapless::String<1024> {
+            core::str::FromStr::from_str(s).expect("Failed to create test string")
+        }
+
+        fn from_str_test<'de, T: serde::Deserialize<'de>>(
```

Comment:
This is copy-pasted from above. Maybe just move it out into the `mod test` mod level.

Reply:
I wanted to be able to see at a glance what `from_str_test` does.

If it was a long function, yes I agree, but as it's a single line, is logically trivial, and is only used in two places, I figured that the local clarity was worth the code duplication.